src/main/java/org/ibp/api/java/impl/middleware/study/validator/StudyValidator.java

```diff
@@ -117,18 +122,42 @@ public void validateStudyShouldNotHaveObservation(final Integer studyId) {
 	}
 
 	public void validateHasNoCrossesOrSelections(final Integer studyId) {
-		this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
+		this.errors = new MapBindingResult(new HashMap<>(), Integer.class.getName());
 		if (this.studyService.hasCrossesOrSelections(studyId)) {
-			errors.reject("study.has.crosses.or.selections");
-			throw new ApiRequestValidationException(errors.getAllErrors());
+			this.errors.reject("study.has.crosses.or.selections");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
 	}
 
 	public void validateStudyHasNoMeansDataset(final Integer studyId) {
-		this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
+		this.errors = new MapBindingResult(new HashMap<>(), Integer.class.getName());
 		if (this.studyService.studyHasGivenDatasetType(studyId, DatasetTypeEnum.MEANS_DATA.getId())) {
-			errors.reject("study.has.means.dataset");
-			throw new ApiRequestValidationException(errors.getAllErrors());
+			this.errors.reject("study.has.means.dataset");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	public void validateDeleteStudy(final Integer studyId) {
+		this.errors = new MapBindingResult(new HashMap<>(), Integer.class.getName());
+		final Study study = this.studyDataManager.getStudy(studyId, false);
+
+		if (study == null) {
+			this.errors.reject("study.id.not.exists", new String[] {studyId.toString()}, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		if (StringUtils.isBlank(study.getProgramUUID())) {
+			this.errors.reject("study.template.delete.not.permitted");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		final Integer studyUserId = study.getUser();
+		final WorkbenchUser user = this.securityService.getCurrentlyLoggedInUser();
+		if (studyUserId != null && !studyUserId.equals(user.getUserid())) {
```

Comment:
If using UserDto, you can replace this if by !user.getUserId().equals(studyUserId)

Reply:
done.